File: admin/src/app/models/create-requests/category-create-request.ts
import { BaseCreateRequest, CategoryDetailCreateRequest } from "@app/models";
import { AutoMap } from "@automapper/classes";

export class CategoryCreateRequest extends BaseCreateRequest {
    @AutoMap({ typeFn: () => CategoryDetailCreateRequest })
    details?: CategoryDetailCreateRequest[];
    @AutoMap()
    parentId?: number = null;
}

File: App.Models/DTOs/Categories/CategoryCreateRequest.cs
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;
using System.Collections.Generic;
using System.ComponentModel;
using App.Models.DTOs.CreateRequests;
using App.Models.DTOs.CategoryDetails;
using App.Common.Model.DTOs;

namespace App.Models.DTOs.Categories
{
    public class CategoryCreateRequest : BaseCreateRequest
    {
        [JsonProperty(PropertyName = "details")]
        [FromForm(Name = "details")]
        public List<CategoryDetailCreateRequest> CategoryDetails { get; set; }
        public int? ParentId { get; set; }
    }
}
